### ISSUE TITLE
dt <= dtmin Aborting. There is either an error in your model specification or the true solution.

### ISSUE THREAD

**Issue by moradza** [SciML/DiffEqFlux.jl#435]
Hi All,
I am new to Julia, and I went through examples and I tried to solve a toy problem  described with the following ode, (I am not sure if there is a user group for the DiffEqFlux so I am asking it here)
I appreciate any help with might be wrong, (the optimal value of p should be 0.0 based on another method)
```
using DifferentialEquations, Plots
using DiffEqSensitivity, OrdinaryDiffEq, Zygote
using DiffEqFlux, OrdinaryDiffEq, Flux, Optim, Plots
using Sundials

u0 = [0.0; 1.0;0.0;0.0;0.0;0.0]
p = [2.]
tspan = (0.0, 10000.0)
function f(du, u,p,t)
    Q1 = 1.0
    Q2 = 1.0
    QM = 2.0
    du[1] = u[2]
    du[2] = -u[2]*u[3]/Q1 - 4* (u[1] - p[1])
    du[3] = u[2]*u[2]-1.0 -u[3]*u[4]/Q2
    du[4] = u[3]*u[3]/Q1 - 1.0 -u[4]*u[5]/Q2
    du[5] = u[4]*u[4]/Q2 - 1.0 - u[5]*u[6]/QM
    du[6] = u[5]*u[5]/Q2 - 1.0
end
prob = ODEProblem(f,u0,tspan,p, dt=0.01)
sol = solve(prob, Tsit5(),saveat=0.1)
tsteps = 0.1
function loss(p)
  sol = solve(prob, Tsit5(), p=p, saveat = tsteps)
  loss = sum(abs2, sol[1,:])
  println(" p : ", p[1], " loss : ", loss)
  return loss, sol
end
result_ode = DiffEqFlux.sciml_train(loss, p,
                                    ADAM(0.1),
                                    maxiters = 100)

remade_solution = solve(remake(prob, p = result_ode.minimizer))
println(result_ode.minimizer)
```

**Comment by ChrisRackauckas** [comment]
It's correctly telling you that the gradient is infinite. You can check via a completely different method (forward mode AD)

```julia
using DifferentialEquations, Plots
using DiffEqSensitivity, OrdinaryDiffEq, Zygote
using DiffEqFlux, OrdinaryDiffEq, Flux, Optim, Plots
using Sundials

u0 = [0.0; 1.0;0.0;0.0;0.0;0.0]
p = [2.]
tspan = (0.0, 10000.0)
function f(du, u,p,t)
    Q1 = 1.0
    Q2 = 1.0
    QM = 2.0
    du[1] = u[2]
    du[2] = -u[2]*u[3]/Q1 - 4* (u[1] - p[1])
    du[3] = u[2]*u[2]-1.0 -u[3]*u[4]/Q2
    du[4] = u[3]*u[3]/Q1 - 1.0 -u[4]*u[5]/Q2
    du[5] = u[4]*u[4]/Q2 - 1.0 - u[5]*u[6]/QM
    du[6] = u[5]*u[5]/Q2 - 1.0
end
prob = ODEProblem(f,u0,tspan,p)
sol = solve(prob, Tsit5(),saveat=0.1)
tsteps = 0.1
function loss(p)
  sol = solve(prob, Tsit5(), p=p, saveat = tsteps)
  loss = sum(abs2, sol[1,:])
  println(" p : ", p[1], " loss : ", loss)
  return loss, sol
end

using ForwardDiff
ForwardDiff.gradient(x->loss(x)[1],p)

1-element Array{Float64,1}:
 -1.783377087217341e154
```

Is the system chaotic? If so, this isn't well-defined.

**Comment by moradza** [comment]
Yes, it is sort of chaotic! I think there is no way to go around this problem, am I right?

**Comment by ChrisRackauckas** [comment]
It's just the wrong method. The solution isn't differentiable w.r.t. parameters since a small perturbation leads to an O(1) change due to chaos... hence the infinite derivative. Try ReservoirComputing.jl if you want something that can predict chaotic dynamics.
